tests/Unit/Helpers/Domain/CoordinateMaps/TestMapHelpers.hpp

```diff
@@ -528,6 +528,106 @@ void test_suite_for_map_on_sphere(
   test_helper(map2);
 }
 
+/*!
+ * \ingroup TestingFrameworkGroup
+ * \brief Given a Map `map`, tests the map functions, including map inverse,
+ * jacobian, and inverse jacobian, for a series of points.
+ * These points are chosen in a right cylinder with cylindrical radius 1
+ * and I1-axis extending from -1 to +1. The
+ * map is expected to be valid on the boundary of that cylinder as well as
+ * in its interior.
+ * This test works only in 3 dimensions.
+ */
+template <typename Map>
+void test_suite_for_map_on_cylinder(const Map& map,
+                                    const double inner_radius,
+                                    const double outer_radius) noexcept {
+  static_assert(Map::dim == 3, "Works only for a 3d map");
+
+  // Set up random number generator
+  MAKE_GENERATOR(gen);
+
+  std::uniform_real_distribution<> radius_dis(inner_radius, outer_radius);
+  std::uniform_real_distribution<> phi_dis(0.0, 2.0 * M_PI);
+  std::uniform_real_distribution<> height_dis(-1.0, 1.0);
+
+  const double height = height_dis(gen);
+  CAPTURE_PRECISE(height);
+  const double phi = phi_dis(gen);
+  CAPTURE_PRECISE(phi);
+  const double radius = radius_dis(gen);
+  CAPTURE_PRECISE(radius);
+
+  const std::array<double, 3> random_point{
+      {radius * cos(phi), radius * sin(phi), height}};
+
+  const std::array<double, 3> random_bdry_point_rho{
+      {outer_radius * cos(phi), outer_radius * sin(phi), height}};
+
+  const std::array<double, 3> random_bdry_point_z{
+      {radius * cos(phi), radius * sin(phi), height > 0.5 ? 1.0 : -1.0}};
+
+  const std::array<double, 3> random_bdry_point_corner{
+      {outer_radius * cos(phi), outer_radius * sin(phi),
+       height > 0.5 ? 1.0 : -1.0}};
+
+  // If inner_radius is zero, this point is on the axis.
+  const std::array<double, 3> random_inner_bdry_point_or_origin{
+      {inner_radius * cos(phi), inner_radius * sin(phi), height}};
+
+  // If inner_radius is zero, this point is on the axis.
+  const std::array<double, 3> random_inner_bdry_point_corner{
+      {inner_radius * cos(phi), inner_radius * sin(phi),
+       height > 0.5 ? 1.0 : -1.0}};
+
+  const auto test_helper = [
+    &random_bdry_point_rho, &random_bdry_point_z, &random_bdry_point_corner,
+    &random_inner_bdry_point_or_origin, &random_inner_bdry_point_corner,
+    &random_point
+  ](const auto& map_to_test) noexcept {
+    test_serialization(map_to_test);
+    CHECK_FALSE(map_to_test != map_to_test);
+
+    test_coordinate_map_argument_types(map_to_test,
```

Comment:
This all looks reasonable, but maybe condense with a loop somehow, since you're just running the same 5 lines over and over again? Maybe store the points in a std::array and then loop over points in the array? Unless there's some performance reason, that seems a bit nicer to me